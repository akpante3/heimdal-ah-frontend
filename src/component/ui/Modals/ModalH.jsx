import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Modal, Button } from 'react-bootstrap';
import { removeModal } from '../../../actions/modals.js';


// render() {
const ModalH = props => (
  (
    <div className={props.classname}>
      <Button bsStyle="primary" bsSize="large">
        Launch modal
      </Button>
      <Modal show={props.display}>
        <Modal.Header>
          <Modal.Title>{props.title}</Modal.Title>
        </Modal.Header>
        <Modal.Body>{props.children}</Modal.Body>
        <Modal.Footer>
          <Button onClick={removeModal()}>Cancel</Button>
          <Button bsStyle="primary">{props.modalType}</Button>
        </Modal.Footer>
      </Modal>
    </div>
  )
);


ModalH.protoType = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  classname: PropTypes.string.isRequired,
  display: PropTypes.bool.isRequired,
  removeModal: PropTypes.func.isRequired,
};

ModalH.defaultProps = {
  title: '',
  children: '',
  classname: '',
  display: true,
};

const mapStateToProps = state => ({
  display: state.modal.display,
});

const matchDispatchToProps = dispatch => ({
  action: bindActionCreators({ removeModal }, dispatch),
});

export default connect(mapStateToProps, matchDispatchToProps)(ModalH);

// export default ModalH;
