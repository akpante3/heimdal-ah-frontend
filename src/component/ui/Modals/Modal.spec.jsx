/* eslint-disable no-undef */
import React from 'react';
import toJson from 'enzyme-to-json';
import ModalH from './ModalH';

describe('Render <ModalH />', () => {
  it('it should render ModalH component', () => {
    const container = shallow(<ModalH />);
    expect(container.length).toBe(1);
  });

  it('should match the snapshot', () => {
    const tree = shallow(<ModalH />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
