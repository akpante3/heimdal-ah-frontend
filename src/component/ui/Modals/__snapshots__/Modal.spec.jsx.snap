// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <ModalH /> should match the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
